docs/DeveloperGuide.md

```diff
@@ -163,6 +163,60 @@ In order to mark a task as completed, the user keys in a valid command (e.g. `ma
 
 The implementation for marking a task as not done is similar.
 
+### Adding a Lesson
+
+In order to add a Lesson into JARVIS, the user keys in a valid command (e.g. `addmc l/mastery check 1 sd/2022-09-15T20:00 ed/2022-09-15T20:30 si/1 si/2`)
+Parsing of the user input is done and a `AddMasteryCheckCommand` is then generated. (See the sequence diagram for deleting a student in the [Logic component](#logic-component))
+
+The sequence diagram is similar apart from:
+1. the command executed and parsed (`addmc l/mastery check 1 sd/2022-09-15T20:00 ed/2022-09-15T20:30 si/1 si/2` instead of `deletestudent 2`)
+2. the different command class (`AddMasteryCheckCommandParser` and `AddMasteryCheckCommand` instead of `DeleteStudentCommandParser` and `DeleteStudentCommand`)
+3. function called in main (`addLesson()` instead of `deleteStudent`)
+
+More detailed implementation of the parsing of user input is be shown in the activity diagram below.
+
+<img src="images/ParseMasteryCheckActivityDiagram.png" width="550"/>
```

Comment:
I think the activity diagram is kind of unnecessary since the parsing basically just checks that every input is valid. The text explanation below should be good enough.

We can just say that `MasteryCheckCommandParser` checks that ..., ... are all valid and throws a `ParseException` otherwise.